src/worker/ingestion/hooks.ts

```diff
@@ -0,0 +1,225 @@
+        }
+    } else if (tokenParts[0] == 'event') {
+        if (tokenParts[1] == 'name') {
+            text = markdown = stringify(event.event)
+        } else if (tokenParts[1] == 'properties' && tokenParts.length > 2) {
+            const propertyName = tokenParts[2]
+            const property = event.properties?.[propertyName]
+            text = markdown = stringify(property)
+        }
+    } else {
+        throw new Error()
+    }
+    return [text, markdown]
+}
+
+export function getFormattedMessage(
+    action: Action,
+    event: PluginEvent,
+    person: Person | undefined,
+    siteUrl: string,
+    webhookType: WebhookType
+): [string, string] {
+    const messageFormat = action.slack_message_format || '[action.name] was triggered by [user.name]'
+    let messageText: string
+    let messageMarkdown: string
+
+    try {
+        const [tokens, tokenizedMessage] = getTokens(messageFormat)
+        const values: string[] = []
+        const markdownValues: string[] = []
+
+        for (const token of tokens) {
+            const tokenParts = token.match(/\w+/g) || []
+
+            const [value, markdownValue] = getValueOfToken(action, event, person, siteUrl, webhookType, tokenParts)
+            values.push(value)
+            markdownValues.push(markdownValue)
+        }
+        messageText = format(tokenizedMessage, ...values)
+        messageMarkdown = format(tokenizedMessage, ...markdownValues)
+    } catch (error) {
+        const [actionName, actionMarkdown] = getActionDetails(action, siteUrl, webhookType)
+        messageText = `⚠ Error: There are one or more formatting errors in the message template for action "${actionName}".`
+        messageMarkdown = `*⚠ Error: There are one or more formatting errors in the message template for action "${actionMarkdown}".*`
+    }
+
+    return [messageText, messageMarkdown]
+}
+
+export class HookCannon {
+    db: DB
+    teamManager: TeamManager
+    organizationManager: OrganizationManager
+    statsd: StatsD | undefined
+
+    constructor(db: DB, teamManager: TeamManager, organizationManager: OrganizationManager, statsd?: StatsD) {
+        this.db = db
+        this.teamManager = teamManager
+        this.organizationManager = organizationManager
+        this.statsd = statsd
+    }
+
+    public async findAndFireHooks(
+        event: PluginEvent,
+        person: Person | undefined,
+        siteUrl: string,
+        actionMatches: Action[]
+    ): Promise<void> {
+        if (!actionMatches.length) {
+            return
+        }
+
+        const team = await this.teamManager.fetchTeam(event.team_id)
+
+        if (!team) {
+            return
+        }
+
+        const webhookUrl = team.slack_incoming_webhook
+        const organization = await this.organizationManager.fetchOrganization(team.organization_id)
+
+        if (webhookUrl) {
+            const webhookRequests = actionMatches
+                .filter((action) => action.post_to_slack)
```

Comment:
I'm a bit confused here. It seems like `slack_incoming_webhook` is used for teams and discord too?

But then, if I filter on `post_to_slack`, is that set to `True` for teams and discord too?

Because otherwise, we aren't posting webhooks for teams / discord ?

Reply:
Not sure I understand, do you mean the specific services Slack, Teams, Discord? You should completely disregard the `slack` part in `slack_incoming_webhook` and `post_to_slack` names, that's only left over from ole days when exclusively Slack was supported.

Reply:
I was aware of this legacy naming, but I guess worth adding in a little comment for people who are new to the codebase so they don't get confused.

Even better would be refactoring it :D